cluster/src/test/scala/org/apache/spark/sql/policy/RestrictTableCreationPolicyTest.scala

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2017 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.sql.policy
+
+import java.sql.{Connection, DriverManager, SQLException}
+import java.util.Properties
+
+import com.pivotal.gemfirexd.internal.engine.Misc
+import com.pivotal.gemfirexd.internal.iapi.error.StandardException
+import com.pivotal.gemfirexd.{Attribute, TestUtil}
+import com.pivotal.gemfirexd.security.{LdapTestServer, SecurityTestUtils}
+import io.snappydata.{Constant, Property, SnappyFunSuite}
+import io.snappydata.core.Data
+import org.scalatest.{BeforeAndAfter, BeforeAndAfterAll}
+import org.junit.Assert.{assertEquals, assertFalse, assertTrue}
+
+import org.apache.spark.{Logging, SparkConf}
+import org.apache.spark.sql.catalyst.expressions.{EqualTo, Literal}
+import org.apache.spark.sql.catalyst.plans.logical.Filter
+import org.apache.spark.sql.types.StringType
+import org.apache.spark.sql.{SaveMode, SnappyContext, SnappySession}
+import org.apache.spark.unsafe.types.UTF8String
+
+class RestrictTableCreationPolicyTest extends SnappyFunSuite
+    with Logging
+    with BeforeAndAfter
+    with BeforeAndAfterAll {
+  val user1 = "gemfire1"
+  val user2 = "gemfire2"
+  val user3 = "gemfire3"
+  val tableCreator = user1
+  val ownerLdapGroup = "gemGroup1" // users gem1, gem2, gem3
+  val otherLdapGroup = "gemGroup3" // users gem6, gem7, gem8
+  val otherUser = "gemfire6"
+  val props = Map.empty[String, String]
+  val schema = "tax"
+  val numElements = 100
+  val colTableName: String = s"$schema.ColumnTable"
+  val rowTableName: String = s"$schema.RowTable"
+  var ownerContext: SnappyContext = _
+
+  private val sysUser = "gemfire10"
+  var serverHostPort: String = _
+
+
+  override def beforeAll(): Unit = {
+    System.setProperty("snappydata.RESTRICT_TABLE_CREATION", "true")
+    this.stopAll()
+    super.beforeAll()
```

Comment:
I think the property should be set after stopAll() to safe (in case stop depends on presence/absence of property).